test/src/testRpcClient.cpp

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <csignal>
+#include <spdlog/spdlog.h>
+#include <up-client-zenoh-cpp/client/upZenohClient.h>
+#include <up-cpp/transport/builder/UAttributesBuilder.h>
+#include <up-cpp/uri/serializer/LongUriSerializer.h>
+#include <gtest/gtest.h>
+
+using namespace uprotocol::utransport;
+using namespace uprotocol::v1;
+using namespace uprotocol::uri;
+using namespace uprotocol::rpc;
+using namespace uprotocol::client;
+
+UUri rpcUri = LongUriSerializer::deserialize("/test_rpc.app/1/rpc.handler");
+UUri rpcNoServerUri = LongUriSerializer::deserialize("/test_rpc.app/1/rpc.noServer");
+
+class RpcServer : public UListener {
+
+     public:
+
+        UStatus onReceive(UMessage &message) const override {
+
+            UStatus status;
+
+            status.set_code(UCode::OK);
+            
+            UAttributesBuilder builder(message.attributes().source(),
+                                       message.attributes().id(), 
+                                       UMessageType::UMESSAGE_TYPE_RESPONSE, 
+                                       UPriority::UPRIORITY_CS0);
+
+            UAttributes responseAttributes = builder.build();
+
+            UPayload outPayload = message.payload();
+
+            UMessage respMessage(outPayload, responseAttributes);
+
+            if (nullptr != message.payload().data()) {
+
+                std::string cmd(message.payload().data(), message.payload().data() + message.payload().size());
+
+                if ("No Response" != cmd) {
+                    return upZenohClient::instance()->send(respMessage);
```

Comment:
We should avoid calls to Instance where we assume we'll have one.   It is entirely possible that instance will return a NULL pointer and we should check for that case.

Reply:
I think that i already fixed it